google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java

```diff
@@ -121,6 +121,7 @@ public static long getApiMaxRequestBytes() {
 
   private static final Logger logger = LoggerFactory.getLogger(Publisher.class);
 
+  private final TopicName topicName;
   private final String topic;
```

Comment:
Maybe rename the String to "cachedTopicNameString" to indicate that the authoritative data source is topicName.